nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -230,7 +231,13 @@ public String installInspectionHook(NetworkElement inspectedPortElement,
         checkArgument(portChain != null,
                       "Cannot find %s by id: %s!", "Service Function Chain", inspectionPortElement.getElementId());
 
-        FlowClassifier flowClassifier = this.utils.buildFlowClassifier(inspectedPortElement.getElementId());
+        Port defaultGatewayInterfacePort = this.utils.fetchDefaultGatewayPort(inspectedPortElement.getElementId());
```

Comment:
this should be done at the OSC level...the install inspection hooks should just be able to install the hook for the provided element and should not have logic for figuring out the gateway etc.

Reply:
Initially I thought of doing that and since it is a workaround i did not want to touch OSC-core part. And this also require changes in sdk ie sdn-controller-api.

Reply:
ahh..k..got it...we are just getting the source port because its required by SFC...i would add a TODO or FIXME to indicate this is a limitation of SFC that source port is required.

i thought we were just ignoring the inspectedPortElement and using only the gateway port...which is not the case.